File: DAY TWO/PYTHON/Book Suggestion System/book_system_test.py
import unittest

import book_system

class TestThatBookSuggestionFunctionWorks(unittest.TestCase):
    
    def test_that_suggested_book_is_from_the_list(self):
        books = ["Solo", "SuperIntelligence", "India After Ghandi", "Dream Count", "Gaza", "Ripple", "hidden Genius", "The Attributes"];
        suggested_book = book_system.get_suggested_books(books)
        self.assertIsNotNone(suggested_book)
        print("Suggested book", suggested_book)
              
    def test_that_the_pagenumber_is_between_one_and_hundred(self):
        pagenumber = book_system.get_pagenumber()
        self.assertTrue(pagenumber >= 1)
        self.assertTrue(pagenumber <= 100)
        print("PageNumber", pagenumber)
       
    def test_that_one_book_is_added(self):
       books = ["Solo", "SuperIntelligence", "India After Ghandi", "Dream Count", "Gaza", "Ripple"];
       new_book = "Hidden Genius"    
       actual =  book_system.add_books(books, new_book)
       result = ["Solo", "SuperIntelligence", "India After Ghandi", "Dream Count", "Gaza", "Ripple", "Hidden Genius" ];
       self.assertEqual(actual, result)
       
       
    def test_that_one_book_is_removed(self):
       books = ["Solo", "SuperIntelligence", "India After Ghandi", "Dream Count", "Gaza"];
       remove_book = "SuperIntelligence"    
       actual = book_system.remove_books(books, remove_book)
       result = ["Solo", "India After Ghandi", "Dream Count", "Gaza"];
       self.assertEqual(actual, result)
       
       
    def test_that_book_name_is_updated_after_changes(self):
       books = ["Solo", "SuperIntelligence", "India After Ghandi", "Dream Count", "Animal Farm", "Gaza"];
       old_bookname = "Animal Farm"
       updated_bookname = "Animal Kingdom"
       actual = book_system.update_books(books, old_bookname, updated_bookname)
       result = ["Solo", "SuperIntelligence", "India After Ghandi", "Dream Count", "Animal Kingdom", "Gaza"];
       self.assertEqual(actual, result)
       
       
    def test_that_all_books_is_shown(self):
       books = ["Solo", "SuperIntelligence", "India After Ghandi", "Dream Count", "Gaza", "Ripple", "hidden Genius", "Option B", "The Attributes", "Animal Farm"];   
       actual = book_system.show_books(books)
       result = ["Solo", "SuperIntelligence", "India After Ghandi", "Dream Count", "Gaza", "Ripple", "hidden Genius", "Option B", "The Attributes", "Animal Farm"]
       self.assertEqual(actual, result)
            
    
        

File: DAY TWO/JAVA/Book Suggestion System/BookSystemTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class BookSystemTest{

    @Test
    public void testThatSuggestedBookIsFromTheList(){
    String bookChoice = BookSystem.getSuggestedBooks();
    assertNotNull(bookChoice);
    System.out.println("Suggested Book: " + bookChoice);
    }     
    
    @Test
    public void testThatPageNumberIsBetweenOneAndHundred(){
    int pageNumber = BookSystem.getPageNumber();
    assertTrue(pageNumber >= 1);
    assertTrue(pageNumber <= 100);
    System.out.println("Page Number: " + pageNumber);
    }
    
    @Test
    public void testThatANewBookIsAdded(){
    String newBook = "Hidden Genius";
    String[] actual = BookSystem.addBooks(newBook);
    String[] result = {"Solo", "SuperIntelligence", "India After Ghandi", "Dream Count", "Gaza", "Ripple", "Option B", "The Attributes", "Animal Farm", "Hidden Genius"};
    assertArrayEquals(actual, result);
    }
    
    @Test
    public void testThatABookIsremoved(){
    String bookName = "Solo";
    String[] actual = BookSystem.removeBooks(bookName);
    String[] result = {"SuperIntelligence", "India After Ghandi", "Dream Count", "Gaza", "Ripple", "Option B", "The Attributes", "Animal Farm"};
    assertArrayEquals(actual, result);
    }
    
    @Test
    public void testThatBookNameIsUpdatedAfterChanges(){
    String oldBookName = "Animal Farm";
    String newBookName = "Animal Kingdom";
    String[] actual = BookSystem.updateBooks(oldBookName, newBookName);
    String[] result = {"Solo", "SuperIntelligence", "India After Ghandi", "Dream Count", "Gaza", "Ripple", "Option B", "The Attributes", "Animal Kingdom"};
    assertArrayEquals(actual, result);
    }
    
    @Test
    public void testThatBooksIsShown(){
    String[] actual = BookSystem.showBooks();
    String[] result = {"Solo", "SuperIntelligence", "India After Ghandi", "Dream Count", "Gaza", "Ripple", "Option B", "The Attributes", "Animal Kingdom"};
    assertArrayEquals(actual, result);
    }      
}
